### PR TITLE
match translation request language 

### PR DESCRIPTION
**Pull request by michaeltorbert**
Issue #2444  #2455

## Proposed changes

We reach out to translate.wordpress.org and get the info for translated locales, such as the percent translated. Due to issues with the way WordPress handles naming the locales, our code wasn't able to distinguish between formal and informal versions of the same language, resulting in incorrect information displaying. 

## Types of changes

What types of changes does your code introduce?

- Bugfix (non-breaking change which fixes an issue)
- Improves existing functionality
- Improves existing code

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. If creating a separate issue for an item, link to the issue # at the end of the line._

- [x] I've selected the appropriate branch (ie x.y rather than master).
- [x] I've created an appropriate title, descriptive of what the PR does.
- [x] Travis-ci runs with no errors.
- [ ] I have added tests that prove my fix is effective/my feature works or have created an issue for it (if appropriate).

## Testing instructions
- Verify the existing behavior. Check out German, German Formal, and others, and make sure the percentage shown is correct (100% locales should not show up at all).
- I've changed a string so that German won't be 100% anymore for testing purposes. Once this has been tested and code reviewed, we need to change it back.

## Further comments

Here are German, German (formal), and French. Note the translation percentages for each: 99%, 82%, and 95%, respectively.
Screen Shot 2019-05-16 at 3 09 02 PM
Screen Shot 2019-05-16 at 3 09 18 PM
Screen Shot 2019-05-16 at 3 10 54 PM

Here is each loaded into the site in the same order. Notice that for German, it shows the info for German (formal), and for German (informal), it doesn't show anything. French is included as a control.
 
Screen Shot 2019-05-16 at 3 09 43 PM
Screen Shot 2019-05-16 at 3 08 47 PM
Screen Shot 2019-05-16 at 3 10 49 PM

### CODE REVIEW

**Comment by wpsmort** [comment]
I found one problem where, if you're language is set to German Formal, the link in our About box goes to https://translate.wordpress.org/projects/wp-plugins/all-in-one-seo-pack/dev/de/default/ when it should go to https://translate.wordpress.org/projects/wp-plugins/all-in-one-seo-pack/dev/de/formal/

**Comment by wpsmort** [comment]
This is tested and is good.

File: inc/translations.php [hunk 1]
```diff
@@ -75,38 +69,58 @@ private function get_locale_data() {
 		 *
 		 * @since 2.3.5
 		 *
-		 * @param $locales
+		 * @param array $locales All locale info for AIOSEOP from translate.wordpress.org.
```
Comment (EkoJR) [inline inc/translations.php]: Nice to see documentation added. It's just the incorrect format with WP Documentation Standards.

`var` is also intended for class variables. For array params it is `type`

https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-1-parameters-that-are-arrays

File: inc/translations.php [hunk 2]
```diff
@@ -75,38 +69,58 @@ private function get_locale_data() {
...
+		 * @var string $formal Indication of whether currently active locale is formal.
+		 * @var string $slug Indication of whether locale from translate.wordpress.org is formal.
+		 * @var string $current_locale Currently active locale.
 		 */
 		private function set_current_locale_data( $locales ) {
 
+			$current_locale = $this->current_locale;
+
+			if ( strpos( $current_locale, '_formal' ) ) {
+				$formal = $this->formal = 'formal';
```
Comment (EkoJR) [inline inc/translations.php]: This is being flagged as `phpcs Squiz.PHP.DisallowMultipleAssignments.Found`, and is also one of the current excluded rules.

**Pull request merged**